### DISCUSSION TITLE
谷歌云盘挂载配置问题

### DISCUSSION THREAD

**Discussion by yuanying1199** [libsgh/PanIndex#353] [Q&A] [answered]
PanIndex版本: 3.1.0，按照[PanIndex Tool](https://pt.noki.icu/) 教程第4步创建客户端凭据，应用类型：Web 应用，已获授权的重定向 URI
：https://pt.noki.icu/ ，然后按步骤可以获得refresh_token。

但添加谷歌云盘后，开流量中转，相应目录无法访问，查日志有400记录；
不开流量中转，302跳转 https://www.googleapis.com/drive/v3/files/?alt=media ，网页显示：

> {"error":{"errors":[{"domain":"global","reason":"unsupportedOutputFormat","message":"Unsupported`` Output Format","locationType":"parameter","location":"alt"}],"code":400,"message":"Unsupported Output Format"}}。

请问那出错？

**Comment by libsgh** [comment]
应该是google api配置那里有问题，仔细检查下

**Reply by yuanying1199** [reply]
google api应该没问题呀，创建凭据，应用类型：Web 应用，已获授权的重定向 URI
：https://pt.noki.icu/ ，已获授权的 JavaScript 来源 尝试加上: 我的域名 http://localhost:5238 https://pt.noki.icu 。OAuth 同意屏幕测试和发布都试了不行，问题还一样存在

那个图文教程不配套，搞不清楚google api哪有问题，给个详细教程对比下最好
我看在 rclone 和 [alist](https://github.com/alist-org/alist) 中应用类型用的是 **桌面设备**，不需要其他参数，可以正常使用。
建议也改成这种方式

**Reply by libsgh** [reply]
> google api应该没问题呀，创建凭据，应用类型：Web 应用，已获授权的重定向 URI ：https://pt.noki.icu/ ，已获授权的 JavaScript 来源 尝试加上: 我的域名 http://localhost:5238 https://pt.noki.icu 。OAuth 同意屏幕测试和发布都试了不行，问题还一样存在
> 
> 那个图文教程不配套，搞不清楚google api哪有问题，给个详细教程对比下最好 我看在 rclone 和 [alist](https://github.com/alist-org/alist) 中应用类型用的是 **桌面设备**，不需要其他参数，可以正常使用。 建议也改成这种方式

应用类型是无所谓的，你按照我写的这个教程再试一次：https://github.com/libsgh/PanIndex/discussions/359

**Reply by yuanying1199** [reply]
问题已经解决了，[#359](https://github.com/libsgh/PanIndex/discussions/359)里边就是我回复的
根目录ID不能留空，是**必填**的，我原来一直留空以为就是网盘根目录没有填root，填上根目录ID就不报错了

**Comment by yuanying1199** [comment, marked as answer]
按教程[#359](https://github.com/libsgh/PanIndex/discussions/359)第9步，挂载Google Drive时**根目录ID不能留空**，必须填上相关ID ( 挂载网盘根目录设置成root )
